File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.py
cardCnt = input()
cards = list(map(int, input().split()))

cmpCnt = input()
cmps = list(map(int, input().split()))
cmpDict = {}

for i in cmps:
    cmpDict[i] = 0

for i in cards:
    if i in cmpDict:
        cmpDict[i] += 1

for i in cmps:
    print(cmpDict[i], end=' ')

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.java
import java.util.*;
import java.io.*;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		HashMap<String, Integer> map = new HashMap<>();
		int cardTotalCnt = Integer.parseInt(br.readLine());
		
		String[] cards = br.readLine().split(" ");
		for (String card : cards) {
			if (map.containsKey(card)) {
				int cardCnt = map.get(card);
				map.replace(card, cardCnt+1);
			} else {
				map.put(card, 1);
			}
		}
		
		int resultCnt = Integer.parseInt(br.readLine());
		String[] results = br.readLine().split(" ");
		for (String result : results) {
			if (map.containsKey(result)) {
				bw.write(map.get(result) + " ");
			} else {
				bw.write("0 ");
			}
		}
		bw.flush();
		bw.close();	
	}
}